tests/integration-tests.test.ts

```diff
@@ -592,4 +592,90 @@ describe('integration tests', () => {
 
    });
 
+   describe('building routes with router.route', () => {
+
+      it('is chainable', () => {
+         let handler = (_req: Request, resp: Response): void => { resp.send('Test'); },
+             getSpy = spy(handler),
+             postSpy = spy(handler),
+             putSpy = spy(handler);
+
+         app.route('/test')
+            .get(getSpy)
+            .post(postSpy)
+            .put(putSpy);
+
+         // Ensure that chained handlers were registered properly
+
+         testOutcome('GET', '/test', 'Test');
+         assert.calledOnce(getSpy);
+         assert.notCalled(postSpy);
+         assert.notCalled(putSpy);
+         getSpy.resetHistory();
+
+         testOutcome('POST', '/test', 'Test');
+         assert.calledOnce(postSpy);
+         assert.notCalled(getSpy);
+         assert.notCalled(putSpy);
+         postSpy.resetHistory();
+
+         testOutcome('PUT', '/test', 'Test');
+         assert.calledOnce(putSpy);
+         assert.notCalled(getSpy);
+         assert.notCalled(postSpy);
+         postSpy.resetHistory();
+      });
+
+      it('registers route handlers properly', () => {
+         let handlers: SinonSpy[] = [],
+             methods: (keyof IRoute & keyof IRouter)[],
+             allHandler: SinonSpy,
+             route: IRoute;
+
+         route = app.route('/test');
+
+         // methods to test
+         methods = [ 'get', 'post', 'put', 'delete', 'head', 'options', 'patch' ];
+
+         // Register handler that runs for every request
+         allHandler = spy((_req: Request, _resp: Response, next: NextCallback) => { next(); });
+         route.all(allHandler);
+
+
+         // Register a handler for each method
+         _.each(methods, (method) => {
+            let handler = spy((_req: Request, resp: Response) => { resp.send('Test'); });
+
+            // Save the handler spy for testing later
+            handlers.push(handler);
+
+            route[method](handler);
+         });
+
+         app.use((_req: Request, resp: Response) => {
+            resp.send('not found');
+         });
+
+         // Run once for each method type
+         _.each(methods, (method) => {
+            testOutcome(method.toUpperCase(), '/test', 'Test');
+         });
+
+         // Check that each handler was called exactly once
+         _.each(handlers, (handler) => { assert.calledOnce(handler); });
+
+         // Check that the "all" handler was called for each method
+         expect(allHandler.callCount).to.strictlyEqual(methods.length);
```

Comment:
I think we could refactor these checks to be inside the `testOutcome` each ... otherwise we're not really testing that the right handler was called on each outcome.